Add SEOScoreAggregator

igorschoester opened the pull request:
## Summary

This PR can be summarized in the following changelog entry:

* [non-user-facing] Adds a SEOScoreAggregator to be used in the TreeAssessor.

## Relevant technical choices:

* Uses the same calculation as `src/assessor.js`.
* Rearranges the files under `tree/assess` and creates index files for each folder.

## Test instructions

This PR can be tested by following these steps:

It is not actually used yet. You could compare scores of the current assessor with the scores of this implementation.

But short of that:
* Look at the code.
* Run the tests.

Fixes #2124

Commit:
Rearranged tree assess files and added index files for each folder

On branch 2124-create-seo-score-aggregator
Changes to be committed:
	renamed:    spec/tree/assess/AssessmentSpec.js -> spec/tree/assess/assessments/AssessmentSpec.js
	renamed:    spec/tree/assess/ScoreAggregatorSpec.js -> spec/tree/assess/scoreAggregators/ScoreAggregatorSpec.js
	new file:   src/tree/assess/assessments/index.js
	modified:   src/tree/assess/index.js
	renamed:    src/tree/assess/ScoreAggregator.js -> src/tree/assess/scoreAggregators/ScoreAggregator.js
	new file:   src/tree/assess/scoreAggregators/index.js

Commit:
Add SEOScoreAggregator

Using the same scoring as in the current assessor (src/assessor.js)

On branch 2124-create-seo-score-aggregator
Changes to be committed:
	new file:   spec/tree/assess/scoreAggregators/SEOScoreAggregatorSpec.js
	new file:   src/tree/assess/scoreAggregators/SEOScoreAggregator.js
	modified:   src/tree/assess/scoreAggregators/index.js

Commit:
Fix failing tests due to index change

hansjovis commented on src/tree/assess/scoreAggregators/index.js:
```diff
@@ -0,0 +1,2 @@
+export { default as ScoreAggregator } from "./ScoreAggregator";
```
Oh, nice syntax. I should remember this.

hansjovis commented on src/tree/assess/scoreAggregators/SEOScoreAggregator.js:
```diff
@@ -0,0 +1,54 @@
+ * @type {number}
+ * @const
+ *
+ * @memberOf module:tree/assess
```
I am not sure if this is needed, since it is a local variable that is not exported.
Let's discuss this when you have the time!

igorschoester replied:
As discussed with you. We document private things too. And even though this is not on the class directly, it is still a part of it.

hansjovis commented on src/tree/assess/index.js:
```diff
@@ -1,6 +1,6 @@
-import ScoreAggregator from "./ScoreAggregator";
-import Assessment from "./assessments/Assessment";
+import * as ScoreAggregators from "./scoreAggregators";
+import * as Assessments from "./assessments";
```
Hmmm, need to up my import game 😛.

igorschoester commented:
Still want to change the tests here, see https://github.com/Yoast/YoastSEO.js/pull/2176/files#r261361029

Commit:
Improve the SEOScoreAggregator spec

Moving the aggregator instantiation into a before each.
Adding a second mix spec that comes from the original assessor test.

hansjovis reviewed (approved):
CR: 👍

Commit:
Merge branch 'develop' of github.com:Yoast/YoastSEO.js into 2124-create-seo-score-aggregator

Changed the imports here.
Something to note: do not let the import of the base class go through an index file.
If you do, the order of exports will matter.
Which is why they are now done with direct imports.

hansjovis commented:
Acceptance test: ✅
